cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -69,23 +63,55 @@ func osbuildStagesToRPMs(stages []osbuild.StageResult) []koji.RPM {
 	return rpms
 }
 
+func appendTargetError(res *worker.OSBuildJobResult, err error) {
+	errStr := err.Error()
+	fmt.Printf("target errored: %s", errStr)
+	res.TargetErrors = append(res.TargetErrors, errStr)
+}
+
 func (impl *OSBuildJobImpl) Run(job worker.Job) error {
-	outputDirectory, err := ioutil.TempDir(impl.Output, job.Id().String()+"-*")
-	if err != nil {
-		return fmt.Errorf("error creating temporary output directory: %v", err)
+	// Initialize variable needed for reporting back to osbuild-composer.
+	var osbuildJobResult *worker.OSBuildJobResult = &worker.OSBuildJobResult{
+		Success: false,
+		OSBuildOutput: &osbuild.Result{
+			Success: false,
+		},
+		UploadStatus: "failure",
 	}
+
+	var outputDirectory string
+
+	// In all cases it is necessary to report result back to osbuild-composer worker API.
 	defer func() {
-		err := os.RemoveAll(outputDirectory)
+		err := job.Update(osbuildJobResult)
+		if err != nil {
+			log.Printf("Error reporting job result: %v", err)
+		}
+
+		err = os.RemoveAll(outputDirectory)
 		if err != nil {
 			log.Printf("Error removing temporary output directory (%s): %v", outputDirectory, err)
 		}
 	}()
 
+	outputDirectory, err := ioutil.TempDir(impl.Output, job.Id().String()+"-*")
+	if err != nil {
+		return fmt.Errorf("error creating temporary output directory: %v", err)
+	}
+
+	// Read the job specification
 	var args worker.OSBuildJob
 	err = job.Args(&args)
 	if err != nil {
 		return err
 	}
+	// The specification allows multiple upload targets because it is an array, but we don't support it.
+	// Return an error to osbuild-composer.
+	if len(args.Targets) > 1 {
+		log.Printf("The job specification contains more than one upload target. This is not supported any more. " +
+			"This might indicate a deployment of incompatible osbuild-worker and osbuild-composer versions.")
+		return nil
```

Comment:
I wonder whether it would not make more sense to create the above comment as an `error` and add it to target errors?

Reply:
I'm not sure really. I think this would make sense as the `InternalError`, but I don't want to add it as part of this PR.